hudi-cli/src/main/java/org/apache/hudi/cli/commands/MetadataCommand.java

```diff
@@ -152,72 +163,192 @@ public String stats() throws IOException {
+      return "[ERROR] Metadata Table not enabled/initialized\n\n";
+    }
 
-    return out.toString();
+    HoodieMetadataConfig fsConfig = HoodieMetadataConfig.newBuilder().enable(false).build();
+    HoodieBackedTableMetadata fsMetaReader = new HoodieBackedTableMetadata(
+        new HoodieLocalEngineContext(HoodieCLI.conf), fsConfig, HoodieCLI.basePath, "/tmp");
+
+    HoodieTimer timer = new HoodieTimer().startTimer();
+    List<String> metadataPartitions = metaReader.getAllPartitionPaths();
+    List<String> fsPartitions = fsMetaReader.getAllPartitionPaths();
+    Collections.sort(fsPartitions);
+    Collections.sort(metadataPartitions);
+
+    Set<String> allPartitions = new HashSet<>();
+    allPartitions.addAll(fsPartitions);
+    allPartitions.addAll(metadataPartitions);
+
+    LOG.info("All FS partitions count " + fsPartitions.size() + ", metadata partition count " + metadataPartitions.size());
+    LOG.info("Partitions equality " + fsPartitions.equals(metadataPartitions));
+
+    if (!fsPartitions.equals(metadataPartitions)) {
+      LOG.error("All FS partitions " + Arrays.toString(fsPartitions.toArray()));
+      LOG.error("All Metadata partitions " + Arrays.toString(metadataPartitions.toArray()));
+    }
+
+    final List<Comparable[]> rows = new ArrayList<>();
+    for (String partition : allPartitions) {
+      Map<String, FileStatus> fileStatusMap = new HashMap<>();
+      Map<String, FileStatus> metaFileStatusMap = new HashMap<>();
+      FileStatus[] metadataStatuses = metaReader.getAllFilesInPartition(new Path(HoodieCLI.basePath, partition));
+      Arrays.stream(metadataStatuses).forEach(entry -> metaFileStatusMap.put(entry.getPath().getName(), entry));
+      FileStatus[] fsStatus = fsMetaReader.getAllFilesInPartition(new Path(HoodieCLI.basePath, partition));
+      Arrays.stream(fsStatus).forEach(entry -> fileStatusMap.put(entry.getPath().getName(), entry));
+
+      Set<String> allFiles = new HashSet<>();
+      allFiles.addAll(fileStatusMap.keySet());
+      allFiles.addAll(metaFileStatusMap.keySet());
+
+      for (String file : allFiles) {
+        Comparable[] row = new Comparable[6];
+        row[0] = partition;
+        FileStatus fsFileStatus = fileStatusMap.get(file);
+        FileStatus metaFileStatus = metaFileStatusMap.get(file);
+        row[1] = file;
+        row[2] = fsFileStatus != null;
+        row[3] = metaFileStatus != null;
+        row[4] = (fsFileStatus != null) ? fsFileStatus.getLen() : 0;
+        row[5] = (metaFileStatus != null) ? metaFileStatus.getLen() : 0;
+        if (verbose) {
+          rows.add(row);
+        }
+      }
+
+      LOG.info("Validating partition " + partition);
+      LOG.info(" total FS Files count " + metadataStatuses.length + ", metadata files count " + fsStatus.length);
+      if (metadataStatuses.length != fsStatus.length) {
+        LOG.error(" Files equality size failed ");
+      }
+
+      for (Map.Entry<String, FileStatus> entry : fileStatusMap.entrySet()) {
```

Comment:
I was referring to these code blocks. Please check if this gives us any additional benefit compared to what we print using HoodiePrintHelper

Reply:
Here we are printing only the ones that are not matching